volttron/platform/vip/agent/core.py

```diff
@@ -429,28 +429,40 @@ def __init__(self, owner, address=None, identity=None, context=None,
         self.address = address
         self.identity = str(identity) if identity is not None else str(uuid.uuid4())
         self.agent_uuid = agent_uuid
-        # The public and secret keys are obtained by:
-        # 1. publickkey and secretkey parameters to __init__
-        # 2. in the query string of the address parameter to __init__
-        # 3. from the agent's keystore
+        self.publickey = publickey
+        self.secretkey = secretkey
+        self.serverkey = serverkey
+        self.developer_mode = developer_mode
 
-        if publickey is None or secretkey is None:
-            publickey, secretkey = self._get_keys()
-        if publickey and secretkey and serverkey:
-            self._add_keys_to_addr(publickey, secretkey, serverkey)
+        self._set_keys()
 
-        if publickey is None:
-            _log.debug('publickey is None')
-        if secretkey is None:
-            _log.debug('secretkey is None')
```

Comment:
should probabably add a line so that it's easy to figure out that you are within the core class.